src/main/java/org/folio/circulation/domain/notice/schedule/ScheduledDueDateNoticeHandler.java

```diff
@@ -0,0 +1,128 @@
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.domain.notice.NoticeContextUtil;
+import org.folio.circulation.domain.notice.NoticeTiming;
+import org.folio.circulation.domain.notice.PatronNoticeService;
+import org.folio.circulation.domain.policy.LoanPolicyRepository;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+
+import io.vertx.core.json.JsonObject;
+
+public class ScheduledDueDateNoticeHandler {
+
+  public static ScheduledDueDateNoticeHandler using(Clients clients, DateTime systemTime) {
+    return new ScheduledDueDateNoticeHandler(
+      new LoanRepository(clients),
+      new LoanPolicyRepository(clients),
+      new ConfigurationRepository(clients),
+      PatronNoticeService.using(clients),
+      ScheduledNoticesRepository.using(clients),
+      systemTime);
+  }
+
+  private LoanRepository loanRepository;
+  private LoanPolicyRepository loanPolicyRepository;
+  private ConfigurationRepository configurationRepository;
+  private PatronNoticeService patronNoticeService;
+  private ScheduledNoticesRepository scheduledNoticesRepository;
+  private DateTime systemTime;
+
+  public ScheduledDueDateNoticeHandler(
+    LoanRepository loanRepository, LoanPolicyRepository loanPolicyRepository,
+    ConfigurationRepository configurationRepository,
+    PatronNoticeService patronNoticeService,
+    ScheduledNoticesRepository scheduledNoticesRepository, DateTime systemTime) {
+
+    this.loanRepository = loanRepository;
+    this.loanPolicyRepository = loanPolicyRepository;
+    this.configurationRepository = configurationRepository;
+    this.patronNoticeService = patronNoticeService;
+    this.scheduledNoticesRepository = scheduledNoticesRepository;
+    this.systemTime = systemTime;
+  }
+
+  public CompletableFuture<Result<Collection<ScheduledNotice>>> handleNotices(Collection<ScheduledNotice> scheduledNotices) {
+    CompletableFuture<Result<ScheduledNotice>> future = completedFuture(succeeded(null));
+    for (ScheduledNotice scheduledNotice : scheduledNotices) {
+      future = future.thenCompose(r -> r.after(v -> handleNotice(scheduledNotice)));
+    }
+    return future.thenApply(r -> r.map(v -> scheduledNotices));
+  }
+
+  private CompletableFuture<Result<ScheduledNotice>> handleNotice(ScheduledNotice notice) {
+    if (notice.getLoanId() != null) {
+      return handleDueDateNotice(notice);
+    }
+    return completedFuture(succeeded(notice));
+  }
+
+  private CompletableFuture<Result<ScheduledNotice>> handleDueDateNotice(ScheduledNotice notice) {
+    return loanRepository.getById(notice.getLoanId())
+      .thenApply(r -> r.map(LoanAndRelatedRecords::new))
+      .thenCompose(r -> r.after(loanPolicyRepository::lookupLoanPolicy))
+      .thenCompose(r -> r.after(configurationRepository::lookupTimeZone))
```

Comment:
What is the purpose of using time zone here?

Reply:
It is supposed to be used to send dates in context with appropriate time zone.
It will become not necessary when dates are localized in mod-template-engine.